### PR TITLE
[FREELDR] Update ACPI Extended Attributes support, to 6.3

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Update BIOS_MEMORY_MAP and PcMemGetBiosMemoryMap(), to ACPI 6.3 from 6.2-A.

### CODE REVIEW

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 1]
```diff
@@ -333,16 +333,34 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
...
                  PcBiosMapCount);
             /* This unlikely case was correct between ACPI 3.0 and 4.0, so assume all is fine.
              * Unless we would be ready to drop ACPI 3.0 compatibility.
              */
             goto nextRange;
         }
 
+        if (((PBIOS_MEMORY_MAP)BIOSCALLBUFFER)->ExtendedAttributes.NonVolatile_Deprecated_Reserved != 0)
+        {
+            WARN("Ignoring deprecated/invalid NonVolatile flag. (PcBiosMapCount = %lu)\n",
```
Comment (learn-more) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: I don't think we need more dprints that don't add any value

Comment (SergeGautherie) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: I agree they are of low value, yet I would like to have this code as a reminder.
Would adding an "#ifdef acpi_future", so it is not compiled, be acceptable?

Comment (HBelusca) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: Just use a comment instead of a DPRINT.

Comment (SergeGautherie) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: If `WARN()` is commented out, then this code doesn't do anything anymore,
hence the suggestion to use a global `#ifdef` (or `/* ... */`), around this code as is.

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 2]
```diff
@@ -333,16 +333,34 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             break;
         }
 
-        if (((PBIOS_MEMORY_MAP)BIOSCALLBUFFER)->ExtendedAttributes.Enabled_Reserved == 0)
+        if (((PBIOS_MEMORY_MAP)BIOSCALLBUFFER)->ExtendedAttributes.Enabled_Reserved != 1)
```
Comment (tkreuzer) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: Please keep the previous. != 1 should never be used, and this doesn't make it anymore readable.

Comment (SergeGautherie) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: Reverted.

**Pull request merged**